Homework24

dhuzhvii opened the pull request.

dhuzhvii committed:
Homework16

RegistationNavigation

dhuzhvii committed:
Homework 17

dhuzhvii committed:
Homework17

add song to playlist

dhuzhvii committed:
Update HW17.java

dhuzhvii committed:
Homework 18

dhuzhvii committed:
Homework19

delete Playlist

dhuzhvii committed:
Homework20

Remove playlist

dhuzhvii committed:
Homework21

Change playlist name with double click

dhuzhvii committed:
Homework22

Please need some help with error massage. Seems like can not find css locator or syntax of hoverOver method is wrong?

dhuzhvii committed:
Homework23

Refactored Login Test and Homework18(playSongTest2)

dhuzhvii committed:
HW24

Implemented Browser Factory method

Berichicko commented:
Please finish the homework!

![image](https://github.com/dhuzhvii/A53-QA-Automation-Framework/assets/104761482/75b632e6-d3d8-4bcd-adbf-e3dc576a095d)

HINT

Hint 

_You can run Chrome browser  or Firefox using your local Grid:_


**Step 1** **You don't have files (selenium-server-4.12.1.jar) in your project A51-QA-Automation-Framework**

For example

**Add the**  **selenium-server-4.12.1.jar**  file to the project
Download Remote Selenium WebDriver (Grid)  https://www.selenium.dev/downloads/

![image](https://github.com/rhieya416/A49-QA-Automation-Framework/assets/104761482/ce97b866-2dca-4fe2-9383-fed5992b0c13)

![image](https://github.com/rhieya416/A49-QA-Automation-Framework/assets/104761482/b8e5132e-219f-4fa3-aa17-9a245f259463)


**Step 2**

To start the selenium server:** `java -jar selenium-server-4.12.1.jar standalone --selenium-manager true`

![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/051e9832-d0a8-4d05-a1c3-a4b9b59e7b7c)

**Step 3**

Add an URL to your project, for example

![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/a9d0805a-89e4-4e3d-b4b9-303c2903d019)

`String gridURL = " http://192.168.0.248:4444";`

```
 public WebDriver pickBrowser(String browser, Map<String, String> variantCapabilities) throws MalformedURLException {
        DesiredCapabilities caps = new DesiredCapabilities(); // Create a new DesiredCapabilities object
        String gridURL = "http://192.168.0.248:4444"; // Set the URL for the Selenium Grid

        if (variantCapabilities != null) {
            caps.merge(new DesiredCapabilities(variantCapabilities)); // Merge any variant capabilities into the DesiredCapabilities object
        }

        switch (browser) {
            case "firefox":
                WebDriverManager.firefoxdriver().setup(); // Set up the FirefoxDriver executable using WebDriverManager
                FirefoxOptions firefoxOptions = new FirefoxOptions(); // Create a new FirefoxOptions object
                firefoxOptions.addPreference("dom.webnotifications.enabled", false); // Disable web notifications
                return new FirefoxDriver(firefoxOptions.merge(caps)); // Launch Firefox with the specified options and capabilities

            case "MicrosoftEdge":
                WebDriverManager.edgedriver().setup(); // Set up the EdgeDriver executable using WebDriverManager
                EdgeOptions edgeOptions = new EdgeOptions(); // Create a new EdgeOptions object
                edgeOptions.addArguments("--remote-allow-origins=*"); // Allow remote origins
                return new EdgeDriver(edgeOptions.merge(caps)); // Launch Edge with the specified options and capabilities

            case "grid-edge":
                caps.setCapability("browserName", "MicrosoftEdge"); // Set the browser capability to Microsoft Edge
                return new RemoteWebDriver(URI.create(gridURL).toURL(), caps); // Launch a remote WebDriver instance with the specified capabilities

            case "grid-firefox":
                caps.setCapability("browserName", "firefox"); // Set the browser capability to Firefox
                return new RemoteWebDriver(URI.create(gridURL).toURL(), caps); // Launch a remote WebDriver instance with the specified capabilities

            case "grid-chrome":
                caps.setCapability("browserName", "chrome"); // Set the browser capability to Chrome
                return new RemoteWebDriver(URI.create(gridURL).toURL(), caps); // Launch a remote WebDriver instance with the specified capabilities

            default:
                WebDriverManager.chromedriver().setup(); // Set up the ChromeDriver executable using WebDriverManager
                ChromeOptions options = new ChromeOptions(); // Create a new ChromeOptions object
                options.addArguments("--disable-notifications", "--remote-allow-origins=*", "--incognito", "--start-maximized"); // Disable notifications, allow remote origins, launch in incognito mode, and start maximized
                options.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"}); // Exclude the enable-automation switch
                return new ChromeDriver(options.merge(caps)); // Launch Chrome with the specified options and capabilities
        }
    }
```

**Step 4**


![image](https://github.com/rhieya416/A49-QA-Automation-Framework/assets/104761482/5ee2eff9-2231-478e-b7a7-199fb0a4e2f1)

![image](https://github.com/rhieya416/A49-QA-Automation-Framework/assets/104761482/120a6812-8217-459f-af51-55c780def394)

Run command in console: `gradle clean test -Dbrowser=grid-firefox`

![image](https://github.com/rhieya416/A49-QA-Automation-Framework/assets/104761482/837fff1d-dbd4-4919-96fc-04b6f5c651fa)


![image](https://github.com/rhieya416/A49-QA-Automation-Framework/assets/104761482/dfaba966-2f48-465c-8f5e-2da897e3222f)

**GRID MacOS ( video)**  https://drive.google.com/file/d/1Tkk_mve-lTeBd4-iSs83OWyXNu0ruCtB/view?usp=sharing

[SeleniumGrid_forMac.pdf](https://github.com/Test-Pro-Bootcamp/A33-QA-Automation-Framework/files/13388392/SeleniumGrid_forMac.pdf)

**Full code** https://drive.google.com/drive/folders/1DwIrvwx6lZZ5SBV72Ys4ibCmZXpgsy4p?usp=sharing


Hope this at least helps you know where you stand.

dhuzhvii committed:
Homework24

completed homework. Ran Login tests and tests from Homework18 using grid-chrome and grid-firefox. Build successful. Please let me know if anything else needed
